TestHelpers.Tests/MockFileTests.cs

```diff
@@ -269,6 +269,73 @@ public void MockFile_GetAttributeWithIllegalParameter_ShouldThrowOneArgumentExce
         }
 
         [Test]
+        public void MockFile_SetAttributeOfExistingFile_ShouldReplaceExistingFlags()
+        {
+            var filedata = new MockFileData("test")
+            {
+                Attributes = FileAttributes.Normal
+            };
+            var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>
+            {
+                {XFS.Path(@"c:\something\demo.txt"), filedata},
+            });
+
+            fileSystem.File.SetAttributes(XFS.Path(@"c:\something\demo.txt"), FileAttributes.Hidden);
+            var attributes = fileSystem.File.GetAttributes(XFS.Path(@"c:\something\demo.txt"));
+            Assert.That(attributes, Is.EqualTo(FileAttributes.Hidden));
+        }
+
+        [Test]
+        public void MockFile_SetAttributeOfExistingDirectory_ShouldReplaceExistingFlags()
+        {
+            var fileSystem = new MockFileSystem();
+            var path = XFS.Path(@"c:\something");
+            fileSystem.AddDirectory(path);
+            var existingAttributes = fileSystem.File.GetAttributes(path);
+
+            fileSystem.File.SetAttributes(path, existingAttributes | FileAttributes.Hidden);
+            var attributes = fileSystem.File.GetAttributes(path);
+            Assert.That(attributes, Is.EqualTo(existingAttributes | FileAttributes.Hidden));
+        }
+
+        [Test]
+        [TestCase("", FileAttributes.Normal)]
+        [TestCase("   ", FileAttributes.Normal)]
+        [TestCase(">", FileAttributes.Normal)]
+        public void MockFile_SetAttributeOfExistingDirectoryWithInvalidParameters_ShouldThrowArgumentException(string path, FileAttributes attributes)
+        {
+            var fileSystem = new MockFileSystem();
+            
```

Comment:
If you look at the possible exceptions for SetAttributes it explicitly lists both empty string and strings with only whitespace. So I'm testing both scenarios.